src/Athena/Controllers/api/OfferingController.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            if (id != offering.Id)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to edit {id} but got a model for {offering.Id}");
+            }
+            await offerings.EditAsync(offering);
+        }
+
+        [HttpDelete("{id}")]
+        public async Task DeleteOffering(Guid id)
+        {
+            var offering = await offerings.GetAsync(id);
+            if (offering == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to delete {offering} that does not exist");
+            }
+            await offerings.DeleteAsync(offering);
+        }
+
+        [HttpGet("api/v1/course/{id}/offering/{offeringId}")]
+        public async Task<IEnumerable<Offering>> GetOfferingsForCourseAsync(Course course)
+        {
+            if (course == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to get offering for {course} that doesn't exist");
+            }
+            return (await offerings.GetOfferingsForCourseAsync(course));
+        }
+
+        [HttpPost("api/v1/course/{id}/offering/{offeringId}/meeting")]
+        public async Task AddMeetingAsync(Offering offering, Meeting meeting)
+        {                                       
+            if (meeting == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add meeting {meeting} where meeting does not exist");
+            }
+            await offerings.AddMeetingAsync(offering, meeting);
+        }
+
+        [HttpDelete("api/v1/course/{id}/offering/{offeringId}/meeting")]
+        public async Task RemoveMeetingAsync(Offering offering, Meeting meeting)
+        {
+            if (meeting == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to remove meeting {meeting} where meeting does not exist");
+            }
+            await offerings.RemoveMeetingAsync(offering, meeting);
+        }
+
+        [HttpGet("api/v1/course/{id}/offering/{offeringId}/meeting")]
+        public async Task<IEnumerable<Meeting>> GetMeetingsForOfferingAsync(Offering offering)
+        {
+            if (offering == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to get meetings for {offering} where offering doesn't exist");
+            }
+            return (await meetings.GetMeetingsForOfferingAsync(offering));
+        }
+
+        [HttpPost("api/v1/course/{id}/offering/{offeringId}")]
+        public async Task AddOfferingAsync(Course course, Offering offering)
+        {
+            if (offering == null)
+            {
+                throw new ApiException(HttpStatusCode.BadRequest, $"Tried to add offering {offering} to course {course} that doesn't exist");
+            }
+            await courses.AddOfferingAsync(course, offering);
+        }
+
+        [HttpDelete("api/v1/course/{id}/offering/{offeringId}")]
```

Comment:
Did you intend to implement this here or in a composite controller? If you intend to implement it here you need to use `/api/v1/course...` to override the controller-wide prefix.